cmd/minishift/cmd/start.go

```diff
@@ -154,42 +128,50 @@ func runStart(cmd *cobra.Command, args []string) {
 	}
 
 	//Create the host directories if not present
-	hostDirs := []string{viper.GetString(startFlags.HostConfigDir.Name),
-		viper.GetString(startFlags.HostDataDir.Name), viper.GetString(startFlags.HostVolumeDir.Name),
-		viper.GetString(startFlags.HostPvDir.Name)}
-	err = clusterup.EnsureHostDirectoriesExist(libMachineClient, hostDirs)
-	if err != nil {
-		fmt.Println("Error creating required host directories: ", err)
-		atexit.Exit(1)
+	hostDirs := []string{
+		viper.GetString(startFlags.HostConfigDir.Name),
+		viper.GetString(startFlags.HostDataDir.Name),
+		viper.GetString(startFlags.HostVolumeDir.Name),
+		viper.GetString(startFlags.HostPvDir.Name),
 	}
-
-	ip, err := host.Driver.GetIP()
+	err = clusterup.EnsureHostDirectoriesExist(hostVm, hostDirs)
 	if err != nil {
-		fmt.Println("Error determining host ip: ", err)
-		atexit.Exit(1)
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating required host directories: %v", err))
 	}
 
-	if proxyConfig.IsEnabled() {
-		// once we know the IP, we need to make sure it is not proxied in a proxy environment
-		proxyConfig.AddNoProxy(ip)
-		proxyConfig.ApplyToEnvironment()
+	autoMountHostFolders(hostVm.Driver)
+
+	openShiftVersion := clusterup.DetermineOpenShiftVersion(viper.GetString(startFlags.OpenshiftVersion.Name))
+	clusterUpConfig := &clusterup.ClusterUpConfig{
+		OpenShiftVersion: openShiftVersion,
+		MachineName:      constants.MachineName,
+		Ip:               ip,
+		Port:             constants.APIServerPort,
+		RoutingSuffix:    viper.GetString(startFlags.RoutingSuffix.Name),
+		HostPvDir:        viper.GetString(startFlags.HostPvDir.Name),
+		User:             defaultUser,
+		Project:          defaultProject,
+		KubeConfigPath:   constants.KubeConfigPath,
+		OcPath:           cacheOc(openShiftVersion),
```

Comment:
Sure, I can do that as well. In this case it becomes something like:

       ocPath := cacheOc(openShiftVersion)
	clusterUpConfig := &clusterup.ClusterUpConfig{
		OpenShiftVersion: openShiftVersion,
		MachineName:      constants.MachineName,
		Ip:               ip,
		Port:             constants.APIServerPort,
		RoutingSuffix:    viper.GetString(startFlags.RoutingSuffix.Name),
		HostPvDir:        viper.GetString(startFlags.HostPvDir.Name),
		User:             defaultUser,
		Project:          defaultProject,
		KubeConfigPath:   constants.KubeConfigPath,
		OcPath:           ocPath,
	}

TBH, I prefer this as well. However, I got several other comments where people thing it is "better" to rather inlines a function call into another function call, over having two separate lines.

> maybe the cacheOc should not happen here. It is a function that does too much

it needs to happen here or just before creating the struct. _ocPath_ downloads and caches the _oc_ binary based on the OpenShift version we are after. It returns the path to the cached binary. I am not sure why you think it is doing too much.

That said, I am fine to split this into two lines. Note though, that I still need the openShiftVersion twice.